plugin-server/src/main/ingestion-queues/batch-processing/each-batch.ts

```diff
@@ -31,42 +31,43 @@ export async function eachBatch(
         for (const messageBatch of messageBatches) {
             const batchSpan = transaction.startChild({ op: 'messageBatch', data: { batchLength: messageBatch.length } })
 
-            if (!isRunning() || isStale()) {
-                status.info('🚪', `Bailing out of a batch of ${batch.messages.length} events (${loggingKey})`, {
-                    isRunning: isRunning(),
-                    isStale: isStale(),
-                    msFromBatchStart: new Date().valueOf() - batchStartTimer.valueOf(),
-                })
-                await heartbeat()
-                return
-            }
+            const processingPromises = messageBatch.map((message: KafkaMessage) => eachMessage(message, queue))
+            processingPromises.push(heartbeat()) // Signal to the Kafka cluster that we are actively processing events
+            await Promise.all(processingPromises)
 
-            const lastBatchMessage = messageBatch[messageBatch.length - 1]
-            await Promise.all(messageBatch.map((message: KafkaMessage) => eachMessage(message, queue)))
+            batchSpan.finish()
+        }
 
-            // this if should never be false, but who can trust computers these days
-            if (lastBatchMessage) {
-                resolveOffset(lastBatchMessage.offset)
-            }
+        // Commit offsets once at the end of the batch. We run the risk of duplicates
+        // if the pod is prematurely killed in the middle of a batch, but this allows
+        // us to process events out of order within a batch, for higher throughput.
```

Comment:
could we have some monitoring on the number of dups that are being added to CH, so we can check that graph as we roll all these changes out? Can be dumb and not 100% accurate metric in a cron job that just gets the uuid dup counts percentage over the past hour of inserts or something similar.